source/extensions/common/wasm/wasm.cc

```diff
@@ -104,6 +104,19 @@ const uint8_t* decodeVarint(const uint8_t* pos, const uint8_t* end, uint32_t* ou
 
 } // namespace
 
+// TODO: consider moving this into StatNameSet as it currently requires 3 table lookups to insert a
+// new entry.
+Stats::StatName StatNameTable::intern(absl::string_view name) {
```

Comment:
i need to chat with you about this; I don't think this is the best pattern for this architecture, and you can tell from other uses of StatNameSet in the codebase that this is not what it's designed for.

I'm in a meeting but I"ll ping you in a few minutes

Reply:
I think we have a short term question which is: is this the source of the memory issue that Istio is having and a longer term one which is what is the right approach for Wasm Stats.

I agree that this isn't what we want in the long term.

Reply:
Discussing offline.  Let's see if we can get this in as a patch then work on the long term plan.